language/compiler/ir-to-bytecode/src/compiler.rs

```diff
@@ -749,6 +752,32 @@ fn compile_functions(
         .collect()
 }
 
+fn compile_function_body_impl(context: &mut Context, ast_function: Function_) -> Result<CodeUnit> {
+    Ok(match ast_function.body {
+        FunctionBody::Move { locals, code } => {
+            let m = type_parameter_indexes(&ast_function.signature.type_formals)?;
+            compile_function_body(context, m, ast_function.signature.formals, locals, code)?
+        }
+        FunctionBody::Bytecode { locals, code } => {
+            let m = type_parameter_indexes(&ast_function.signature.type_formals)?;
+            compile_function_body_bytecode(
+                context,
+                m,
+                ast_function.signature.formals,
+                locals,
+                code,
+            )?
+        }
+
+        FunctionBody::Native => {
+            for (var, _) in ast_function.signature.formals.into_iter() {
+                record_src_loc!(local: context, var)
+            }
+            CodeUnit::default()
```

Comment:
Eeek! I really hate `default` for things like this. We should look into removing it at some point

Reply:
Yes that's the one of the two remaining items on my list. The other one is locals and parameters being overlapped.